src/SFA.DAS.Forecasting.Domain.UnitTests/Commitments/EmployerCommitmentTests.cs

```diff
@@ -40,14 +42,64 @@ public void SetUp()
         public void Updates_Existing_Commitment()
         {
             var commitment = _moqer.Resolve<EmployerCommitment>();
-            Assert.IsTrue( commitment.RegisterCommitment(3, "test apprentice", "test course", 1, 4, "test provider", DateTime.Today,
-                DateTime.Today.AddDays(1), null, 87.27m, 240, 12));
-            Assert.AreEqual(commitment.ApprenticeName, "test apprentice");
-            Assert.AreEqual(commitment.ProviderName, "test provider");
-            Assert.AreEqual(commitment.CourseName, "test course");
-            Assert.AreEqual(commitment.ProviderId, 4);
-            Assert.AreEqual(commitment.LearnerId, 3);
-            Assert.AreEqual(commitment.CourseLevel, 1);
+
+            var newCommitment = new CommitmentModel
+            {
+                EmployerAccountId = 1,
+                ApprenticeshipId = 2,
+                LearnerId = 3,
+                ApprenticeName = "test apprentice",
+                ProviderName = "test provider",
+                CourseName = "test course",
+                ProviderId = 4,
+                MonthlyInstallment = 10,
+                NumberOfInstallments = 1,
+                CompletionAmount = 50,
+                CourseLevel = 1,
+                FundingSource = FundingSource.Levy,
+                SendingEmployerAccountId = 55501
+            };
+
+            commitment.RegisterCommitment(newCommitment).Should().BeTrue();
+
+            commitment.ApprenticeName.Should().Be("test apprentice");
+            commitment.ProviderName.Should().Be("test provider");
+            commitment.CourseName.Should().Be("test course");
+            commitment.ProviderId.Should().Be(4);
+            commitment.LearnerId.Should().Be(3);
+            commitment.CourseLevel.Should().Be(1);
+            commitment.CompletionAmount.Should().Be(50);
+            commitment.SendingEmployerAccountId.Should().Be(55501);
+            commitment.FundingSource.Should().Be(FundingSource.Levy);
+        }
+
+        [Test]
+        public void Should_not_update_if_id_0()
+        {
+            var commitment = _moqer.Resolve<EmployerCommitment>();
+
+            _existingCommitment.EmployerAccountId = 0;
+
+            var newCommitment = new CommitmentModel
+            {
+                EmployerAccountId = 1,
+                ApprenticeshipId = 2,
+                LearnerId = 3,
+                ApprenticeName = "test apprentice",
+                ProviderName = "test provider",
+                CourseName = "test course",
+                ProviderId = 4,
+                MonthlyInstallment = 10,
+                NumberOfInstallments = 1,
+                CompletionAmount = 50,
+                CourseLevel = 1,
+                FundingSource = FundingSource.Levy,
+                SendingEmployerAccountId = 55501
+            };
```

Comment:
Repeated from above, would be good to put this in the setup? or in a method to avoid duplication of code